.github/workflows/main.yml

```diff
@@ -207,3 +207,49 @@ jobs:
         with:
           command: clippy
           args: -- -D warnings
+
+  test_coverage:
+    name: Test Coverage
+    needs: update_deps
+    runs-on: ubuntu-latest
+    env:
+      SHA_SHORT: ${GITHUB_SHA:0:8}
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Download Cargo.lock
+        uses: actions/download-artifact@v2
+        with:
+          name: lockfile
+
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+
+      - name: Restore cargo registry index
+        uses: actions/cache@v2
+        with:
+          path: ~/.cargo/registry/index
+          key: cargo-index-v2-${{ needs.update_deps.outputs.crates-io-index-head }}
+
+      - name: Restore dependency crates
+        uses: actions/cache@v2
+        with:
+          path: ~/.cargo/registry/cache
+          key: cargo-deps-v2-${{ hashFiles('Cargo.lock') }}
+
+      - name: Run tarpaulin
+        uses: actions-rs/tarpaulin@v0.1
+        with:
+          out-type: Html
+
+      - run: mv tarpaulin-report.html test_coverage_$SHA_SHORT.html
+
+      - name: Upload coverage report
+        uses: actions/upload-artifact@v2
+        with:
+          name: test_coverage_$SHA_SHORT
```

Comment:
You need to use `${{ env.SHA_SHORT }}` outside of shell scripts.

Reply:
Done, thanks